addons/portal_rating/static/src/js/portal_rating_composer.js

```diff
@@ -32,64 +38,54 @@ var RatingPopupComposer = publicWidget.Widget.extend({
             'display_rating': true,
             'csrf_token': odoo.csrf_token,
             'user_id': session.user_id,
         });
-    },
-    /**
-     * @override
-     */
-    start: function () {
-        var defs = [];
-        defs.push(this._super.apply(this, arguments));
-
-        // instanciate and insert composer widget
-        this._composer = new PortalComposer(this, this.options);
-        defs.push(this._composer.replace(this.$('.o_portal_chatter_composer')));
 
-        return Promise.all(defs);
-    },
-});
-
-publicWidget.registry.RatingPopupComposer = publicWidget.Widget.extend({
-    selector: '.o_rating_popup_composer',
-    custom_events: {
-        reload_rating_popup_composer: '_onReloadRatingPopupComposer',
+        return def;
     },
 
     /**
      * @override
      */
     start: function () {
-        this.ratingPopupData = this.$el.data();
-        this.ratingPopupData.display_composer = !this.ratingPopupData.disable_composer && !session.is_website_user;
-        this.ratingPopup = new RatingPopupComposer(this, this.ratingPopupData);
-        return Promise.all([
-            this._super.apply(this, arguments),
-            this.ratingPopup.appendTo(this.$el)
-        ]);
+        const def = this._super.apply(this, arguments);
+        this._reloadRatingPopupComposer();
+        return def;
     },
 
-    //--------------------------------------------------------------------------
-    // Private
-    //--------------------------------------------------------------------------
-
     /**
      * Destroy existing ratingPopup and insert new ratingPopup widget
      *
      * @private
      * @param {Object} data
      */
-    _reloadRatingPopupComposer: function (data) {
-        if (this.ratingPopup) {
-            this.ratingPopup.destroy();
+    _reloadRatingPopupComposer: function () {
+        if (this.options.hide_rating_avg) {
+            this.$('.o_rating_popup_composer_stars').empty();
+        } else {
+            const ratingAverage = qweb.render('portal_rating.rating_stars_static', {inline_mode: true, widget: this, 'val': this.rating_avg});
+            this.$('.o_rating_popup_composer_stars').empty().html(ratingAverage);
         }
-        if (this.ratingPopupData.display_composer) {
-            this.ratingPopup = new RatingPopupComposer(this, Object.assign(this.ratingPopupData, data));
-            this.ratingPopup.appendTo(this.$el);
+
+        // Append the modal
+        const modal = qweb.render('portal_rating.PopupComposer', {inline_mode: true, widget: this, 'val': this.rating_avg});
+        this.$('.o_rating_popup_composer_modal').html(modal);
+
+        if (this._composer) {
+            this._composer.destroy();
         }
+
+        // Instantiate the "Portal Composer" widget and insert it into the modal
+        this._composer = new PortalComposer(this, this.options);
+        this._composer.appendTo(this.$('.o_rating_popup_composer_modal .o_portal_chatter_composer'));
```

Comment:
Could return the promise of this "appendTo" so your start override can return Promise.all([superPromise, composerPromise]).